src/cluster.c

```diff
@@ -4948,6 +4962,163 @@ void addNodeReplyForClusterSlot(client *c, clusterNode *node, int start_slot, in
     setDeferredArrayLen(c, nested_replylen, nested_elements);
 }
 
+void addNodeDetailsToShardReply(client *c, clusterNode *node) {
+    /*
+     * Following information regarding a node is returned.
+     *
+     * 1. id (40 char len)
+     * 2. port
+     * 3. ip
+     * 4. hostname
+     * 5. endpoint
+     * 5. role
+     * 6. replication offset (in case of replica only).
+     * 7. health
+     */
+    int reply_cnt = 0;
+    void *node_replylen = addReplyDeferredLen(c);
+    addReplyBulkCString(c, "id");
+    addReplyBulkCBuffer(c, node->name, CLUSTER_NAMELEN);
+    reply_cnt++;
+    addReplyBulkCString(c, "port");
+    addReplyLongLong(c, node->port);
+    reply_cnt++;
+    addReplyBulkCString(c, "ip");
+    addReplyBulkCString(c, node->ip);
+    reply_cnt++;
+    if (node->hostname) {
+        addReplyBulkCString(c, "hostname");
+        addReplyBulkCString(c, node->hostname);
+        reply_cnt++;
+    }
+    const char *endpoint = getPreferredEndpoint(node);
+    if (!strcmp(endpoint, "")) {
+        addReplyBulkCString(c, "endpoint");
+        addReplyBulkCString(c, endpoint);
+        reply_cnt++;
+    }
+
+    addReplyBulkCString(c, "role");
+    if (nodeIsSlave(node)) {
+        addReplyBulkCString(c, "replica");
+        addReplyBulkCString(c, "replication-offset");
+        addReplyLongLong(c, node->repl_offset);
+        reply_cnt+=2;
+    } else {
+        addReplyBulkCString(c, "primary");
```

Comment:
Hm, we may need to take a call on this. Although it's used sporadically in documentation, "primary" is not really used as a role delimiter. We probably have to keep using master here for the time being for consistency.

Reply:
One possibility: We don't need "role" if we just decide that the first node is the master and the following nodes are replicas.

Reply:
I think keeping the role is safe, don't want to leave it to the assumption. I'm moving it to `master` for now.